pkg/e2etemplates/aws_operator_chart_values.go

```diff
@@ -20,37 +22,127 @@ const AWSOperatorChartValues = `Installation:
       SSH:
         SSOPublicKey: 'test'
       Update:
-        Enabled: ${GUEST_UPDATE_ENABLED}
+        Enabled: {{ .Guest.Update.Enabled }}
     Name: ci-aws-operator
     Provider:
       AWS:
-        Region: ${AWS_REGION}
+        Region: '{{ .Provider.AWS.Region }}'
         DeleteLoggingBucket: true
         IncludeTags: true
         Route53:
           Enabled: true
+	{{- if .Provider.AWS.Encrypter }}
+        Encrypter: '{{ .Provider.AWS.Encrypter }}'
+	{{- else }}
         Encrypter: 'kms'
+	{{- end }}
     Registry:
       Domain: quay.io
     Secret:
       AWSOperator:
-        IDRSAPub: ${IDRSA_PUB}
+        IDRSAPub: {{ .Secret.AWSOperator.IDRSAPub }}
         SecretYaml: |
           service:
             aws:
               accesskey:
-                id: ${GUEST_AWS_ACCESS_KEY_ID}
-                secret: ${GUEST_AWS_SECRET_ACCESS_KEY}
-                token: ${GUEST_AWS_SESSION_TOKEN}
+                id: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.ID }}
+                secret: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.Secret }}
+                token: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.Token }}
               hostaccesskey:
-                id: ${HOST_AWS_ACCESS_KEY_ID}
-                secret: ${HOST_AWS_SECRET_ACCESS_KEY}
-                token: ${HOST_AWS_SESSION_TOKEN}
+                id: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.ID }}
+                secret: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.Secret }}
+                token: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.Token }}
 
       Registry:
         PullSecret:
-          DockerConfigJSON: "{\"auths\":{\"quay.io\":{\"auth\":\"${REGISTRY_PULL_SECRET}\"}}}"
+          DockerConfigJSON: "{\"auths\":{\"quay.io\":{\"auth\":\"{{ .RegistryPullSecret }}\"}}}"
     Security:
       RestrictAccess:
         Enabled: false
 `
+
+type AWSOperatorChart struct {
+	Guest    AWSOperatorChartGuest
+	Provider AWSOperatorChartProvider
+	Secret   AWSOperatorChartSecret
+
+	RegistryPullSecret string
+}
+
```

Comment:
Here some validation to not waste time during e2e test. Better to spend time once here IMO.

Reply:
Thinking of it during a break:
I should just create `func NewAWSOperatorChartValues(config AWSOperatorChartValuesConfig) (string, error)` and call `Render` inside. Then we would have `invalidConfigError` and all the usual business.

Reply:
Also I think extracting `chartvalues` makes sense here. Then we could just call `chartvalues.NewAWSOpeartor()` and have way shorter names. Also easy to detect for `goimpoarts`.